pycloudlib/azure/instance.py

```diff
@@ -2,19 +2,42 @@
 """Azure instance."""
 
 import time
+from collections import namedtuple
+from enum import Enum, auto
 from typing import List, Optional
 
+import requests
+from azure.core.exceptions import ResourceExistsError
+
 from pycloudlib.errors import PycloudlibTimeoutError
 from pycloudlib.instance import BaseInstance
 
+BootDiagnostics = namedtuple("BootDiagnostics", ["console_log_url", "logs"])
+
+
+class VMInstanceStatus(Enum):
+    """Represents Azure VM state."""
```

Comment:
As I see it, this enum does represent pycloudlib's vision of the instance, not the general state/status of it. I wonder if we can improve the documentation for future us.